### PR TITLE
Fix broken two touch zoom for Attachment Images

### PR DESCRIPTION
**Pull request by Jag96**
@marcaaron will you please review this?

### Details
This PR fixes a regression where zooming with two fingers on an attachment was broken.

### Fixed Issues

$ https://github.com/Expensify/App/issues/2499

### Tests/QA
1. Sign into an account and send an image attachment
2. After the attachment uploads, tap on it, confirm an attachment modal shows
3. Double tap on the image, confirm it zooms
4. Double-tap again, confirm it zooms out
5. Use two fingers to zoom in, confirm the image zooms in
6. While zoomed in, swipe down and right and confirm the modal doesn't close, and that the viewport pans around the zoomed image
7. Double-tap again then swipe down or right, confirm the modal dismisses
8. Confirm that there are no regressions on web
    1. Confirm attachments still show up on web
    2. Confirm the modal attachment closes normally on web

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

### CODE REVIEW

**Review by yuwenmemon — DISMISSED** [review]

File: src/components/ImageView/index.native.js [hunk 1]
```diff
@@ -31,6 +32,28 @@ class ImageView extends PureComponent {
...
+    }
+
+    /**
+     * Handler for the initial tap for the PanResponder on our ImageZoom overlay View
+     *
+     * @param {Event} e
+     * @param {GestureState} gestureState
+     * @returns {Boolean}
+     */
+    handleOnStartShouldSetPanResponder(e, gestureState) {
```
Comment (marcaaron) [inline src/components/ImageView/index.native.js:50]: Style guide rule about this [here](https://github.com/Expensify/App/blob/main/STYLE.md#naming-conventions)

File: src/components/ImageView/index.native.js [hunk 2]
```diff
@@ -52,12 +75,13 @@ class ImageView extends PureComponent {
                     cropHeight={windowHeight}
                     imageWidth={this.state.imageWidth}
                     imageHeight={this.state.imageHeight}
-                    onStartShouldSetPanResponder={(e) => {
+                    onStartShouldSetPanResponder={() => {
                         const isDoubleClick = new Date().getTime() - this.lastClickTime <= this.doubleClickInterval;
+                        const touches = this.amountOfTouches;
```
Comment (marcaaron) [inline src/components/ImageView/index.native.js:80]: NAB, don't really need to declare `touches` variable

File: src/components/ImageView/index.native.js [hunk 3]
```diff
@@ -95,6 +119,20 @@ class ImageView extends PureComponent {
                             this.setState({imageHeight, imageWidth});
                         }}
                     />
+                    {/**
+                     Create an invisible view on top of the image so we can capture and set the amount of touches before
+                     the ImageZoom's PanResponder does. Children will be triggered first, so this needs to be inside the
+                     ImageZoom to work
+                     */}
```
Comment (marcaaron) [inline src/components/ImageView/index.native.js:126]: Whoa, nice workaround

**Review by marcaaron — CHANGES_REQUESTED** [review]
Looks good there's just a small style guide rule we should fix up

**Comment by Jag96** [comment]
Updated!

**Review by marcaaron — APPROVED** [review]
LGTM 🤘

**Pull request merged**